server/server.go

```diff
@@ -103,10 +105,60 @@ type ArgoCDServerOpts struct {
 	RepoClientset   reposerver.Clientset
 }
 
+//defaultSettings sets default secret settings
+func defaultSettings(settingsMgr *settings_util.SettingsManager, opts ArgoCDServerOpts) (*settings_util.ArgoCDSettings, error) {
+	defaultString := ""
+	cdSettings, err := settingsMgr.GetSettings()
+	errors.CheckError(err)
+
+	if bytes.Compare(cdSettings.ServerSignature, []byte(defaultString)) == 0 {
+		// set JWT signature
+		signature, err := util_session.MakeSignature(32)
+		errors.CheckError(err)
+		cdSettings.ServerSignature = signature
+	}
+
+	if cdSettings.LocalUsers[common.ArgoCDAdminUsername] == "" {
+		//placeholder for when we replace this with get pod
+		passwordRaw := "password"
+		hashedPassword, err := password.HashPassword(passwordRaw)
+		errors.CheckError(err)
+		log.Warnf("password set to default, please update")
+		cdSettings.LocalUsers = map[string]string{
+			common.ArgoCDAdminUsername: hashedPassword,
+		}
+	} else if valid, _ := password.VerifyPassword("password", cdSettings.LocalUsers[common.ArgoCDAdminUsername]); valid {
```

Comment:
Rather than checking for a default password, I'm wondering if just using an invalid hash (that can never be validated) might be more secure.  Maybe we could add an accessor to determine if the hash is equal to an asterisk (`== "*"`) to say if the account is disabled, then can just check if the default account is disabled.